Automattic/wp-calypso
# Use ProductIcon from @automattic/components instead of internal components

This PR is a follow-up work to recently deployed #38253 (see: pbtFFM-11-p2).

It replaces all usages of Calypso-specific `ProductIcon` and `PlanIcon` components with a common `@automattic/components` `ProductIcon` component.

#### Changes proposed in this Pull Request

* Introduce a utility `getProductIconSlug()` function for getting a mapping between plan or product slug and icon slug (af2b05f)
* Use common `ProductIcon` in `MyPlanCard` (af2b05f)
* Replace `PlanIcon` with `ProductIcon` in `PlanFeaturesHeader` (4f328d0)
* Replace `PlanIcon` with `ProductIcon` in `Banner` (69c3711) 
* Replace `PlanIcon` with `ProductIcon` in `PurchaseItem` (6901f67)
* Replace `PlanIcon` with `ProductIcon` in `ManagePurchase` (51b0abf)
* Revert some of the changes done in #37448 to `PurchaseItem` and `ManagePurchase` (6901f67 and 51b0abf) since now Jetpack Backup icons are supported by the `ProductIcon`.
* Replace `PlanIcon` with `ProductIcon` in `PlanThankYouCard` (441fd7c)
* Replace `PlanIcon` with `ProductIcon` in `UpgradeNudgeExpanded` (831a537)
* Remove unused `PlanIcon` and `ProductIcon` components (1aa994d  and f056782)

#### Testing instructions

Go to each of the following pages and confirm that the plan icons are displayed as expected (no regression):
* http://calypso.localhost:3000/devdocs/design/banner
<img width="804" alt="Screenshot 2019-12-17 at 17 27 35" src="https://user-images.githubusercontent.com/478735/71014632-885a1180-20f2-11ea-835f-bf71e5664e6e.png">
 
* http://calypso.localhost:3000/devdocs/blocks/plan-thank-you-card
<img width="516" alt="Screenshot 2019-12-17 at 17 26 35" src="https://user-images.githubusercontent.com/478735/71014597-77110500-20f2-11ea-9821-a4fa1ffaa280.png">

* http://calypso.localhost:3000/devdocs/blocks/upgrade-nudge-expanded
<img width="831" alt="Screenshot 2019-12-17 at 17 26 15" src="https://user-images.githubusercontent.com/478735/71014569-6c567000-20f2-11ea-8124-419b4305a334.png">

* http://calypso.localhost:3000/me/purchases
<img width="735" alt="Screenshot 2019-12-17 at 17 25 24" src="https://user-images.githubusercontent.com/478735/71014459-3e712b80-20f2-11ea-92a5-b69b560b9f20.png">

* http://calypso.localhost:3000/me/purchases/[site]/[purchase] (go to individual purchases of various plans and products)
<img width="728" alt="Screenshot 2019-12-17 at 16 27 57" src="https://user-images.githubusercontent.com/478735/71014469-4204b280-20f2-11ea-9a37-41479d293611.png">

* http://calypso.localhost:3000/plans/[site] (small icons in the plans grid on desktop)
<img width="841" alt="Screenshot 2019-12-17 at 16 03 17" src="https://user-images.githubusercontent.com/478735/71014267-f7833600-20f1-11ea-97fe-1937dcd199df.png">

* http://calypso.localhost:3000/plans/my-plan/[site] (icons in the My plan/My Products lists)
<img width="842" alt="Screenshot 2019-12-17 at 16 03 24" src="https://user-images.githubusercontent.com/478735/71014282-feaa4400-20f1-11ea-92d1-54e5d66ac612.png">

Fixes n/a
